From ff1b657f73bad5a26c8a1982391cdf66c51ac71b Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 18 Sep 2024 10:26:55 +0300
Subject: [PATCH] [FSW-9512] fs1: check boot info efuse disabled

With this patch, added boot info disable efuse
bit check. If boot info disabled manifest data
will be used from persistent section otherwise
lfs files read/write used.

Change-Id: I131684c45e7c19138ee8a7b213d32f1c12d19349
---

diff --git a/arch/fs1/arc/dev_boot_info.c b/arch/fs1/arc/dev_boot_info.c
index 95e283d..da4f2aa 100644
--- a/arch/fs1/arc/dev_boot_info.c
+++ b/arch/fs1/arc/dev_boot_info.c
@@ -22,17 +22,15 @@
 
 bool boot_rom_skip_boot_info(void)
 {
-	if (boot_rom_global.die_type != FS1_IO_DIE_MASTER)
-		return true;
-
-	/* TODO: efuse check can be added here */
-
-	return false;
+	return !!(boot_rom_global.config2_fuse && EFUSE_SYSCFG2_DISABLE_BOOT_INFO);
 }
 
 void boot_rom_set_boot_info(void)
 {
 	struct fs1_cold_rst_data cold_rst_data;
+
+	if (!boot_rom_in_master_die())
+		return;
 
 	if(boot_rom_skip_boot_info())
 		return;
@@ -56,6 +54,9 @@
 
 void boot_rom_get_boot_info(void)
 {
+	if (!boot_rom_in_master_die())
+		return;
+
 	if(boot_rom_skip_boot_info())
 		return;
 
